core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -169,6 +170,20 @@ public Expression visitRelevanceFieldList(RelevanceFieldList node, AnalysisConte
         ImmutableMap.copyOf(node.getFieldList())));
   }
 
+  @Override
+  public Expression visitFunctionWithCachedValue(FunctionWithCachedValue node,
+                                                 AnalysisContext context) {
+    var valueName = node.toString();
```

Comment:
I think it's sufficient to use `node.getFuncName` here.

`toString` is not a great option to generate a hashmap key. All it guarantees is that the result is a string -- intended for logging and debugging. There is no requirement that it's unique for different object.

In the case of `Function` object, it happens to include name and parameters making it ok as a key but it would be just as valid for `Function.toString` to return `"An unresolved function expression"` or `"Function object 0xDEADBEEF"`

Reply:
Thank you, fixed in 4008c753.